### ISSUE TITLE
Purge Client Side History

### ISSUE THREAD

**Issue by skyfox675** [jsxc/jsxc#448]
I was thinking of a means of "securely" purging a devices previous message cache. I know that MAM is in the works, which makes me think that something like the follow may be possible:

-When window is out of focus, clear local message cache
-Scrolling up within a chat requests history from MAM

Thanks,
Skyfox675

PS. Thanks for the great app, the OTR Verification helped me a ton the other day!

**Comment by sualko** [comment]
What is the benefit of this behavior? Maybe I don't understand you correctly.

**Comment by skyfox675** [comment]
Sualko, thanks for the quick response! 

So my thought is that once JSXC supports MAM, there is no need to store previous messages on each device. Even when using OTR, previous messages can be read later after a successful login, and before establishing a verified session with a recipient leaving message history available, and in the case that a window is reopened prior to the cookie expiration time, they can be read without an actual login. The ultimate fix would be OMEMO with no client side stored messages but I understand that is a ways out. 

So in short, just proposing that a there be a timed (maybe?) purge of client side history to limit amount history stored on an individual device.

**Comment by sualko** [comment]
Currently only 10 messages are saved on client side, so there should be no urgent need for such a feature.

By the way, we (a student of us) are currently working on an omemo implementation, see #418.
